src/dbcparser.cpp

```diff
@@ -73,6 +84,159 @@ std::string dos2unix(const std::string& data)
+            }
+        }
+        canDb.messages.erase(messageIt->first);
+        canDb.messages[updatedMessage] = existingSignals;
+    }
+}
+
+void setMessageComment(CANdb_t &canDb, uint32_t id, const std::string& comment)
+{
+    cdb_debug("Setting the comment for message {} to \"{}\"", id, comment);
+    bool messageItFound = false;
+    auto messageIt = getMessageIteratorById(canDb, id, messageItFound);
+    if (messageItFound) {
+        // This implementation is by no means optimal, because it requires
+        // creating copies of the message/signal pair already in the map,
+        // updating the message's comment, removing the old data and inserting
+        // the new stuff. This is unfortunately necessary since std::map
+        // requires keys to be immutable, so they cannot be updated directly.
+        cdb_debug("Found the message that needs the new comment");
+        CANmessage updatedMessage(messageIt->first);
+        std::vector<CANsignal> existingSignals(messageIt->second);
+        updatedMessage.comment = comment;
+        canDb.messages.erase(messageIt->first);
+        canDb.messages[updatedMessage] = existingSignals;
+    }
+}
+
+void setSignalComment(CANdb_t &canDb, uint32_t id,
+    const std::string& signalName, const std::string& comment)
+{
+    cdb_debug("Setting the comment for signal {}:{} to \"{}\"", id, signalName,
+        comment);
+    bool signalItFound = false;
+    auto signalIt = getSignalIteratorByMessageIdAndName(canDb, id, signalName,
+        signalItFound);
+    if (signalItFound) {
+        cdb_debug("Found the signal that needs the new comment");
+        signalIt->comment = comment;
+    }
+}
+
+void setMessageCycleTime(CANdb_t &canDb, uint32_t id, std::uint32_t cycleTime)
+{
+    cdb_debug("Setting the cycle time for message {} to \"{}\"", id, cycleTime);
+    bool messageItFound = false;
+    auto messageIt = getMessageIteratorById(canDb, id, messageItFound);
+    if (messageItFound) {
+        // See notes in setMessageComment regarding this copy/remove/insert
+        // implementation.
+        cdb_debug("Found the message that needs the new cycle time");
+        CANmessage updatedMessage(messageIt->first);
+        std::vector<CANsignal> existingSignals(messageIt->second);
+        updatedMessage.cycleTime = cycleTime;
+        canDb.messages.erase(messageIt->first);
+        canDb.messages[updatedMessage] = existingSignals;
+    }
+}
+
+void setSignalStartValue(CANdb_t &canDb, uint32_t id,
+    const std::string& signalName, const boost::any& value)
+{
+    cdb_debug("Setting the start value for signal {}:{}", id, signalName);
+    bool signalItFound = false;
+    auto signalIt = getSignalIteratorByMessageIdAndName(canDb, id, signalName,
+        signalItFound);
+    if (signalItFound) {
+        cdb_debug("Found the signal that needs the new start value");
+        signalIt->startValue = value;
+    }
+}
+
+void setSignalValueType(CANdb_t &canDb, uint32_t id,
```

Comment:
return type bool maybe useful for these "setBlah" functions, especially when showing warnings

Reply:
As discussed on Slack, we'll revisit this later depending on how we want to do warning, error and exception handling (if what we already have isn't enough). But in general, you're right -- that would be a better API.